src/components/TodoInfo.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { FC, useRef, useState } from 'react';
+import { MakeChange } from '../types/MakeChange';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  todo: Todo,
+  setTodos: MakeChange,
+  setChangeTitle: React.Dispatch<React.SetStateAction<number | null>>,
+};
+
+export const TodoInfo: FC<Props> = ({
+  todo,
+  setTodos,
+  setChangeTitle,
+}) => {
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const inputNewTitle = useRef<HTMLInputElement | null>(null);
+  const hendleRemove = (id: number) => {
+    setTodos.remove([id]);
+  };
+
+  const hendleToggler = (currentTodo: Todo) => {
+    setTodos.toggle([{ ...currentTodo, completed: !todo.completed }]);
+  };
+
+  const hendleDoubleClick = (id: number) => {
+    setChangeTitle(id);
+    inputNewTitle.current?.focus();
+  };
+
+  const hendleNewTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(e.target.value);
+  };
+
+  const hendleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      if (!newTitle.trim()) {
+        setTodos.remove([todo.id]);
+        setChangeTitle(null);
+
+        return;
+      }
+
+      setTodos.toggle([{ ...todo, title: newTitle }]);
+      setChangeTitle(null);
+
+      return;
+    }
+
+    if (e.key === 'Escape') {
+      setNewTitle(todo.title);
+      setChangeTitle(null);
+    }
+  };
+
+  const hendleClick = (e: { stopPropagation: () => void; }) => {
```

Comment:
Try to add another type for it
```suggestion
  const hendleClick = (e: ...) => {
```